build-push-container/action.yml

```diff
@@ -0,0 +1,174 @@
+name: "Build and push container image"
+description: "Builds a docker image and pushes it to the github registry, Outputs the generated container image version. Requires Microsoft.NET.Build.Containers package installed."
+
+inputs:
+  githubToken:
+    description: "Access token provided by the calling action context, used to publish the packages and push the corresponding tag."
+    required: true
+  packageReadonlyPat:
+    description: "Personal access token used to access the github private nuget source."
+    required: true
+  projectFolder:
+    description: "Folder containing the project to build."
+    required: true
+    type: string
+  imageName:
+    description: "Name of the docker image to build."
+    required: true
+    type: string
+  imageTagPrefix:
+    description: "Prefix of the docker image tag."
+    required: true
+    type: string
+  imageTagSuffix:
+    description: "Suffix of the docker image tag. If not provided, will load the last commit sha."
+    required: false
+    type: string
+  containerPublishType:
+    description: "Type of container publish (api | worker)."
+    required: true
+    type: string
+  dotnetVersion:
+    description: "Version of dotnet to use. Default is v7.x."
+    required: false
+    type: string
+    default: "7.x"
+  dockerRegistry:
+    description: "Docker registry to push the image to."
+    required: false
+    type: string
+    default: "docker.pkg.github.com"
+  doCheckout:
+    description: "Whether to checkout the repository or not."
+    required: false
+    type: boolean
+    default: true
+
+outputs:
+  tag:
+    description: "Docker Image Tag"
+    value: ${{ steps.setImageTag.outputs.tag}}
+
+runs:
+  using: "composite"
+  steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+      if: ${{inputs.doCheckout == true}}
+
+    - name: .NET Install
+      env:
+        DOTNET_INSTALL_DIR: "./.dotnet" # Required by the runner to install dotnet
+      uses: actions/setup-dotnet@v3
+      with:
+        dotnet-version: ${{ inputs.dotnetVersion }}
+
+    - name: Add Trakx Github Nuget Source
+      shell: bash
+      run: >
+        dotnet nuget add source "https://nuget.pkg.github.com/trakx/index.json"
+        --name "github"
+        --username "trakx-bot"
+        --password ${{inputs.packageReadonlyPat}}
```

Comment:
Same here, I don't remember if there is a good reason not to use GitHubToken

Reply:
I don't know either 🤣 
I'll open a task to check this later.